addons/website_sale/static/src/js/website_sale.js

```diff
@@ -414,56 +414,21 @@ publicWidget.registry.WebsiteSale = publicWidget.Widget.extend(VariantMixin, {
      * @override
      * @private
      */
-    _updateProductImage: function ($productContainer, displayImage, productId, productTemplateId, new_carousel, isCombinationPossible) {
-        var $img;
+    _updateProductImage: function ($productContainer, displayImage, productId, productTemplateId, newCarousel, isCombinationPossible) {
         var $carousel = $productContainer.find('#o-carousel-product');
-
-        if (isCombinationPossible === undefined) {
-            isCombinationPossible = this.isSelectedVariantAllowed;
+        // When using the web editor, don't reload this or the images won't
+        // be able to be edited depending on if this is done loading before
+        // or after the editor is ready.
+        if (window.location.search.indexOf('enable_editor') === -1) {
```

Comment:
We should probably use $.deparam or probably another direct var existing with editor. @qsm-odoo pov ?

Reply:
Not sure to understand the code here, if the editor is enabled I don't think we go through here in the first place since we are in a public widget and they are not enabled in edit mode by default... :) @seb-odoo ?

Reply:
That code was needed for sure in 12.0 but maybe not since the editor changes? 

I suppose we can remove the condition now then.

Reply:
I still have the issue when I remove the condition, here is how to reproduce it:
- add `pdb` in `get_combination_info_website`
- visit product page with `?enable_editor=1`
- `continue` on `pdb` when the editor is ready

-> The RCP will return the carousel, and the method will be called, which will replace the DOM of the carousel. -> The editor will not take this new DOM into account, and the images won't be editable.

Reply:
Mmh... I understand the problem... but maybe we should fix it at the root.

In 12.0, when we go to a page with `?enable_editor=1`, this is what happens:
1) Instantiate all public widgets
2) Begin the edit mode: first destroy all public widgets
3) Instantiate all public widgets again (but their internal `editableMode` attribute will be true and many are declared with "if editableMode do nothing")

In 12.5, it is the same except that public widgets are not started in edit mode at all anymore by default (they must be explicitly declared with `disabledInEditableMode` to `false` if you want to).

Your problem is probably either that:
A) we don't wait for step 1 to be completely finished (promises) before doing step 2
B) your specific widget does not wait properly for the promises to be resolved during its instantiation

If A is true it must be solved in at least 12.0, if B is true it must be solved here I guess. In both cases, in master / 12.5, maybe we should not do step 1 and 2 at all if the URL contains `?enable_editor=1`, this would probably speed up those page loading too.

Reply:
I believe the answer is B:

`this.triggerVariantChange(this.$el);` is not linked to the Promise that is returned in `start`.
Since this is a trigger, it is based on events, and there is no promise to link at all.

To fix it correctly a bigger refactoring of this whole class would be needed, which was planned since I started working with it but it has never been the priority compared to other things (and it is still not today).

The first step of the refactoring would be to make `_getCombinationInfo` independent of `ev` so that it can be called manually without a trigger. Instead of being called once per product, depending on ev, the RPC should be optimized to be callable only one time, by taking the list of all products that have to be computed at once (from the optional products modal).
And that would also allow to clean up the `memoize` and other strange stuff.

Another part of the refactoring is that actually `_getCombinationInfo` should not be called at all in `start`. The controller/view can render the page correctly directly, the only exception that has to be fixed is to compute the out of stock feature, as commented in the code.

The final refactoring is that this whole code should use proper Widget structure instead of one Mixin that is inherited everywhere just to call sometimes one of its method, sometimes another.


If this line is the only concern of the current PR, I propose that we move forward with it, keeping the existing condition for now, and keeping in mind that it can be improved.

Reply:
Fine by me